### ISSUE TITLE
[BUG] Tests are flaky

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-go#568]
### What is the bug?

https://github.com/opensearch-project/opensearch-go/actions/runs/9293150937/job/25575676932

### How can one reproduce the bug?

1 out of 5 PRs fails tests intermittently, retrying works.

### What is the expected behavior?

Tests to pass reliably.

**Comment by dblock** [comment]
https://github.com/opensearch-project/opensearch-go/actions/runs/9665682575/job/26663396843?pr=570

**Comment by sean-** [comment]
And a huge number of fixes related to this are in #787, where tests are now being run in parallel with 0 failures.

**Comment by sean-** [comment]
And #811, #821 and #823 should help.

**Comment by sean-** [comment]
Resolved across #782, #787, #811, #820, #821, #823, #834.

Thanks for filing this and for the patient bug reports across multiple CI runs. The flakes have been chased through a long list of changes - #782 (version bumps), #787 (parallel-test reliability and the bulk of the fixes), #811 (pool replacement), #820 (flaky health check), #821 (empty discovery), #823 (discovery error detail), and #834 (test modernization). CI is running parallel and clean now.

If a flake reappears in a fresh run, please file a new issue with the failing job link so it can be tracked against the current shape of the test suite. Closing this one as resolved.
